full-moon/src/ast/parsers.rs

```diff
@@ -523,12 +542,37 @@ define_parser!(
     ParseGenericFor,
     GenericFor<'a>,
     |_, state: ParserState<'a>| {
-        let (state, for_token) = ParseSymbol(Symbol::For).parse(state)?;
-        let (state, names) = expect!(
-            state,
-            OneOrMore(ParseIdentifier, ParseSymbol(Symbol::Comma), false).parse(state),
-            "expected names"
-        );
+        let (mut state, for_token) = ParseSymbol(Symbol::For).parse(state)?;
+
+        let mut names;
```

Comment:
```suggestion
        let names;
```

Variables that are assigned later, and only once, do not need to be marked mutable.

Reply:
This needs to be labelled `mut` as well because the Punctuated struct is being mutated later when the names are pushed to it